### ISSUE TITLE
[REFACTOR] Release the GIL in DynamoDBClient.ping()

### ISSUE THREAD

**Issue by leandrodamascena** [ferrumio/pydynox#385]
## Description

`DynamoDBClient.ping()` holds the Python GIL while waiting for the `ListTables` response from DynamoDB. All other sync operations (`sync_put_item`, `sync_get_item`, `sync_query`, etc.) already release the GIL via `py.detach()` during the network I/O. `ping` is the only one left holding it.

This means threaded Python apps that call `client.ping()` (health checks, connection probes) block every other Python thread during the round-trip to DynamoDB.

## Relevant code

`src/client/mod.rs` — current implementation:

```rust
pub fn ping(&self) -> PyResult<bool> {
    let client = self.client.clone();
    let result = self
        .runtime
        .block_on(async { client.list_tables().limit(1).send().await });

    match result {
        Ok(_) => Ok(true),
        Err(_) => Ok(false),
    }
}
```

Compare with any other sync op (e.g. `basic_operations/put.rs`):

```rust
let result = py.detach(|| runtime.block_on(execute_put_item(client.clone(), prepared)));
```

## Impact

Small but real. A common pattern is:

```python
# Health check thread
def health_check_loop():
    while True:
        if not client.ping():
            alert()
        time.sleep(30)

# Worker threads handling real requests
def worker():
    user = User.sync_get(pk="USER#1")  # blocked while ping holds the GIL
    ...
```

Every 30 seconds, worker threads pause for the duration of the `ListTables` round-trip (can be 10-100ms depending on region). Not catastrophic, but inconsistent with the rest of the codebase.

## Proposed solution

Wrap the `block_on` with `py.detach()`, same pattern as every other sync op:

```rust
pub fn ping(&self, py: Python<'_>) -> PyResult<bool> {
    let client = self.client.clone();
    let result = py.detach(|| {
        self.runtime
            .block_on(async { client.list_tables().limit(1).send().await })
    });

    match result {
        Ok(_) => Ok(true),
        Err(_) => Ok(false),
    }
}
```

PyO3 injects `Python<'_>` automatically — Python users still call `client.ping()` with no arguments.

## What changes

- `ping()` releases the GIL during the network call
- Threaded apps get real parallelism during health checks
- Behavior consistent with the rest of the sync API

## What does NOT change

- Python API — same signature, same return type
- Behavior for single-threaded code
- Error handling

## Files to change

| File | Action |
|------|--------|
| `src/client/mod.rs` | Wrap `block_on` in `ping()` with `py.detach()`, add `py: Python<'_>` param |

## Tasks

- [ ] Update `ping()` to take `py: Python<'_>` and wrap `block_on` with `py.detach()`
- [ ] Confirm `cargo clippy` passes with zero warnings
- [ ] Confirm `uv run maturin develop` builds
- [ ] Run `uv run pytest tests/unit/ -v` — all pass
- [ ] Run `uv run pytest tests/integration/ -v` — all pass (existing ping tests still work)
